backend/pkg/api/connect/service/quota/v1/service.go

```diff
@@ -0,0 +1,169 @@
+// by the Apache License, Version 2.0
+
+// Package quota contains all handlers for the Quota endpoints.
+package quota
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"sort"
+
+	"connectrpc.com/connect"
+	"github.com/redpanda-data/common-go/api/pagination"
+	"github.com/twmb/franz-go/pkg/kerr"
+
+	"github.com/redpanda-data/console/backend/pkg/config"
+	"github.com/redpanda-data/console/backend/pkg/console"
+	v1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1"
+	"github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1/dataplanev1connect"
+)
+
+var _ dataplanev1connect.QuotaServiceHandler = (*Service)(nil)
+
+// Service implements the handlers for Quota endpoints.
+type Service struct {
+	cfg        *config.Config
+	logger     *slog.Logger
+	consoleSvc console.Servicer
+
+	kafkaClientMapper *kafkaClientMapper
+}
+
+// NewService creates a new Quota service handler.
+func NewService(cfg *config.Config,
+	logger *slog.Logger,
+	consoleSvc console.Servicer,
+) *Service {
+	return &Service{
+		cfg:               cfg,
+		logger:            logger,
+		consoleSvc:        consoleSvc,
+		kafkaClientMapper: &kafkaClientMapper{},
+	}
+}
+
+// ListQuotas lists client quotas based on filter criteria.
+func (s *Service) ListQuotas(ctx context.Context, req *connect.Request[v1.ListQuotasRequest]) (*connect.Response[v1.ListQuotasResponse], error) {
+	kafkaReq := s.kafkaClientMapper.listQuotasRequestToKafka(req.Msg)
+
+	resp, err := s.consoleSvc.DescribeClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to describe quotas: %w", err))
+	}
+
+	// Handle errors from the console service
+	if err = kerr.ErrorForCode(resp.ErrorCode); err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to describe quotas: %s", err))
+	}
+
+	// Map console response to protobuf response
+	quotaEntries, err := s.kafkaClientMapper.quotaItemsToProto(resp.Entries)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to map quota items to proto: %w", err))
+	}
+
+	// Add pagination
+	var nextPageToken string
+	if req.Msg.GetPageSize() > 0 {
+		sort.SliceStable(quotaEntries, func(i, j int) bool {
+			if quotaEntries[i].Entity.EntityType != quotaEntries[j].Entity.EntityType {
+				return quotaEntries[i].Entity.EntityType < quotaEntries[j].Entity.EntityType
+			}
+			return quotaEntries[i].Entity.EntityName < quotaEntries[j].Entity.EntityName
+		})
+		page, token, err := pagination.SliceToPaginatedWithToken(quotaEntries, int(req.Msg.PageSize), req.Msg.GetPageToken(), "quotaEntry", func(x *v1.ListQuotasResponse_QuotaEntry) string {
+			return fmt.Sprintf("%d_%s", x.Entity.EntityType, x.Entity.EntityName)
+		})
+		if err != nil {
+			return nil, connect.NewError(
+				connect.CodeInternal,
+				fmt.Errorf("failed to apply pagination: %w", err),
+			)
```

Comment:
NIT: we try to add a reason on all errors (which can be used by the frontend potentially to differentiate)

```suggestion
			return nil, connect.NewError(
				connect.CodeInternal,
				fmt.Errorf("failed to apply pagination: %w", err),
				apierrors.NewErrorInfo(v1.Reason_REASON_CONSOLE_ERROR.String()),
			)
```